embedded-kafka/src/main/java/com/playtika/test/kafka/KafkaTopicsConfigurer.java

```diff
@@ -5,41 +5,51 @@
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
 import org.testcontainers.containers.Container;
 import org.testcontainers.containers.GenericContainer;
 
 import javax.annotation.PostConstruct;
-
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 import static com.playtika.test.common.utils.ContainerUtils.executeInContainer;
 
 @Slf4j
 @RequiredArgsConstructor
 @Getter
 public class KafkaTopicsConfigurer {
+    private static final int DEFAULT_PARTITION_COUNT = 1;
 
     private final GenericContainer kafka;
     private final ZookeeperConfigurationProperties zookeeperProperties;
     private final KafkaConfigurationProperties kafkaProperties;
+    private final Map<String, Integer> partitionsByTopicsToCreate;
 
     @PostConstruct
     void configure() {
-        createTopics(this.kafkaProperties.getTopicsToCreate());
+        createTopics(this.kafkaProperties.getTopicsToCreate(), partitionsByTopicsToCreate);
         restrictTopics(KafkaConfigurationProperties.KAFKA_USER, this.kafkaProperties.getSecureTopics());
     }
 
-    public void createTopics(Collection<String> topics) {
-        if (!topics.isEmpty()) {
-            log.info("Creating Kafka topics: {}", topics);
-            topics.parallelStream()
-                    .forEach(this::createTopic);
-            log.info("Created Kafka topics: {}", topics);
+    public void createTopics(Collection<String> topics, Map<String, Integer> partitionsByTopicsToCreate) {
+        Map<String, Integer> topicToPartitionMap = topics.stream().collect(Collectors.toMap(topic -> topic, topic -> DEFAULT_PARTITION_COUNT));
+        topicToPartitionMap.putAll(partitionsByTopicsToCreate);
+
+        if (!topicToPartitionMap.isEmpty()) {
+            log.info("Creating Kafka topics with partitions: {}", topicToPartitionMap);
+            topicToPartitionMap.entrySet().parallelStream()
+                               .forEach(this::createTopic);
+            log.info("Created Kafka topics with partitions: {}", topicToPartitionMap);
         }
     }
 
-    private void createTopic(String topic) {
-        String[] createTopicCmd = getCreateTopicCmd(topic, zookeeperProperties.getZookeeperConnect());
+    private void createTopic(Map.Entry<String, Integer> topicToPartitionsEntry) {
```

Comment:
please instead create simple pojo for your needs. map.entry is difficult to read and extend